lib/dentaku/ast/range.rb

```diff
@@ -1,10 +1,29 @@
-require_relative "./literal"
+require_relative "./operation"
 
 module Dentaku
   module AST
-    class Range < Literal
-      def value_type
-        :range
+    class Range < Operation
+      def types
+        [:numeric, :numeric, :range]
+      end
+
+      # must return something responding to ===, in this case a
+      # regular ruby Range.
+      def value
+        (left.value..right.value)
+      end
+
+      def generate_constraints!(context)
+        # pass. this is handled by AST::Case
```

Comment:
and we don't need to do anything for when it's used as an arg in a function?

Reply:
wait, wouldn't this mean that `field:numeric..field:text` passes the type checker?

Reply:
you're absolutely right. i've updated this and added a few more things in following commits.